add benchmark for PT fill_ op

mingzhe09088 opened the pull request:
Summary: add benchmark for #22758, to run this benchmark: 
1. Run: python setup.py install in benchmarks/operator_benchmark/pt_extension 
2. Run the benchmark: python -m pt.fill_test

Differential Revision: D16263458

mingzhe09088 commented:
@xuhdev This PR has the benchmark I mentioned. I highly recommend using it when you are working on other operator level performance optimization.

xuhdev commented:
Thanks! Do I have to write such a script for every operator that I will be working on? Is there a generic script, say, for unary and binary operators?

mingzhe09088 commented:
For unary operator, please look at pt/unary_test.py. You could add your operator there.

xuhdev commented on benchmarks/operator_benchmark/pt/fill_test.py:
```diff
@@ -0,0 +1,34 @@
+    N=[10, 1000],
+    device=['cuda', 'cpu'],
+    dtype=[torch.int8, torch.uint8, torch.int16, torch.int32, 
+           torch.int64, torch.half, torch.float, torch.double],
```
Can we also add bool here? Sorry I didn't add it in my original version.

mingzhe09088 replied:
Added.

xuhdev replied:
This should be `torch.bool` not `torch.ByteTensor`...

mingzhe09088 replied:
Changed in #22913 

On a different note, were you able to run the benchmark on your machine?

xuhdev replied:
Yes, I was able to run the benchmark (only on Python 2). But I found the instructions a bit incomplete (as in `benchmarks/operator_benchmark/README.md`:

- `python -m pt.xxx` should be run in `benchmarks/operator_benchmark/`, otherwise `pt` would not be found.
- `protobuf` should be installed (`pip install protobuf`).
- Python 2 must be used.

mingzhe09088 replied:
Thanks for the feedback. I will modify the README accordingly so others won't encounter the same issue.

xuhdev commented on benchmarks/operator_benchmark/pt/fill_test.py:
```diff
@@ -0,0 +1,34 @@
+
+fill_short_configs = op_bench.cross_product_configs(
+    N=[10, 1000],
+    device=['cuda', 'cpu'],
```
Perhaps `device=torch.testing.get_all_device_types()`?

mingzhe09088 replied:
Yes.

xuhdev commented on benchmarks/operator_benchmark/pt/fill_test.py:
```diff
@@ -0,0 +1,34 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
```
Are these imports needed? (e.g., I don't see print used)

mingzhe09088 replied:
That was just to make internal linter happy. It's ok to remove them.

xuhdev commented:
I followed your instruction but it reports error:

    ModuleNotFoundError: No module named 'past'

Do we have to use python 2 for running benchmarks?

mingzhe09088 committed:
add benchmark for PT fill_ op (#22867)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22867

as title

Differential Revision: D16263458

fbshipit-source-id: 3974f094a9ec454d97e81ec8abbe395cd5ff4388

mingzhe09088 commented:
Yes. I was mainly developing in python2 env.
